src/index/vocabulary/UnicodeVocabulary.h

```diff
@@ -60,6 +72,34 @@ class UnicodeVocabulary {
     return _underlyingVocabulary.upper_bound(word, actualComparator);
   }
 
+  // Same as lower_bound or upper_bound, except that word is known to be a full
+  // word, not a prefix. Special handling may be applied in the presence of a
+  // SplitVocabulary.
+  template <typename T>
+  auto getPositionOfWordComparator(const T&) const {
+    return [this](const auto& a, const auto& b) {
+      return _comparator(a, b, SortLevel::TOTAL);
+    };
+  }
+  template <typename T>
+  WordAndIndex getPositionOfWordLower(const T& word) const {
+    auto actualComparator = getPositionOfWordComparator(word);
+    if constexpr (HasSpecialGetPositionOfWord<UnderlyingVocabulary>) {
+      return _underlyingVocabulary.getPositionOfWordLower(word,
+                                                          actualComparator);
+    }
+    return _underlyingVocabulary.lower_bound(word, actualComparator);
+  }
+  template <typename T>
+  WordAndIndex getPositionOfWordUpper(const T& word) const {
+    auto actualComparator = getPositionOfWordComparator(word);
```

Comment:
Can also be simplified with the single function interface (no upper boudn needed)